### PR TITLE
Bust cache of organization page when new member is added

### LINKED ISSUE #19811: Bust cache of organization page when new member is added
**Describe the bug**

After adding a new member to the organization, sometimes the "Meet the team" area on the organization profile page won't update to reflect the new user being added.

**To Reproduce**

1. Add a user to the org
2. Observe the "Meet the team" section on the Org profile page
3. Notice that the newly-added org member isn't immediately visible
4. Bust the page via `/admin/advanced/tools`
5. See that the newly-added org member is now visible

**Expected behavior**

The newly-added org member is immediately visible in the "Meet the team" section in the Org profile page.

**Additional context**
Referring to this section here on the left:

org profile page

### PR DESCRIPTION
**Pull request by rt4914**
## What type of PR is this? (check all applicable)

- [x] Bug Fix

## Related Tickets & Documents

- Closes #19811

## QA Instructions, Screenshots, Recordings

[Loom Video just for context](https://www.loom.com/share/ac1944bcf5ba471a91df8be0b82a9c0d?sid=b394c84c-b2aa-4925-9077-88e1675a76f9)

### How to reproduce the issue?

We want to add a new member to the organisation and check that organisation's page to see if its immediately getting reflected.

_**All these below mentioned steps won't work locally and in uffizzi. It will work in production code only.**_

**Create a new organisation**
1. Visit `settings/organization`
2. Create a new organisation, maybe with `testorg` name & username
3. Visit `/testorg` page and see how many members are there. (There should be only one member).

**Add user to the organisation**
1. In incognito mode or in a different browser create a new user. (You might need to enable `Email Auth` first in `/admin/customization/config`)
2. Now from the other browser in which you created the org, copy the org secret from `/settings/organization`
3. Paste this code in the "new-user" account, `/settings/organization`

1. Now  again visit the testorg page in your admin/first-user
2. Notice that the newly added user is not visible.

### UI accessibility checklist

No UI change.

## Added/updated tests?

- [x] No

## [optional] Are there any post deployment tasks we need to perform?

@maestromac We will need to deploy this on any production app to test this correctly. Check [loom video](https://www.loom.com/share/ac1944bcf5ba471a91df8be0b82a9c0d?sid=b394c84c-b2aa-4925-9077-88e1675a76f9) for more context.

## [optional] What gif best describes this PR or how it makes you feel?

alt_text

### CODE REVIEW

**Comment by rt4914** [comment]
@maestromac @filleduchaos Looking for a PR review on this.

File: app/models/organization_membership.rb [hunk 1]
```diff
@@ -23,4 +25,10 @@ class OrganizationMembership < ApplicationRecord
   def update_user_organization_info_updated_at
     user.touch(:organization_info_updated_at)
   end
+
+  private
+
+  def bust_cache
+    Organizations::BustCacheWorker.perform_async(organization.id, organization.slug)
```
Comment (filleduchaos) [inline app/models/organization_membership.rb:32]: Hmmm, this busts cache for all the organization's articles as well but that's probably okay?

**Review by filleduchaos — APPROVED** [review]
🎉

File: app/models/organization_membership.rb (hunk 1 above)
Comment (maestromac) [reply app/models/organization_membership.rb:32]: That's a good point @filleduchaos . 

I think busting all org's articles on new member is a bit overkill. I think a good alternative is `BustCachePathWorker.perform_async(organization.path.to_s)`

Comment (benhalpern) [reply app/models/organization_membership.rb:32]: Agreed @maestromac

Comment (rt4914) [reply app/models/organization_membership.rb]: Thanks for the suggestion.

I have updated the code and test accoridngly.

cc: @filleduchaos @maestromac @benhalpern

**Review by maestromac — APPROVED** [review]
LGTM!

**Pull request merged**